services/vision/obstacledistance/obstacle_distance.go

```diff
@@ -0,0 +1,139 @@
+// Package obstacledistance uses an underlying camera to fulfill vision service methods, specifically
+// GetObjectPointClouds, which performs several queries of NextPointCloud and returns a median point.
+package obstacledistance
+
+import (
+	"context"
+
+	"github.com/edaniels/golog"
+	"github.com/golang/geo/r3"
+	"github.com/montanaflynn/stats"
+	"github.com/pkg/errors"
+	"go.opencensus.io/trace"
+
+	"go.viam.com/rdk/components/camera"
+	"go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/robot"
+	svision "go.viam.com/rdk/services/vision"
+	"go.viam.com/rdk/spatialmath"
+	"go.viam.com/rdk/utils"
+	vision "go.viam.com/rdk/vision"
+)
+
+var model = resource.DefaultModelFamily.WithModel("obstacle_distance")
+
+// DistanceDetectorConfig specifies the parameters for the camera to be used
+// for the obstacle distance detection service.
+type DistanceDetectorConfig struct {
+	NumQueries int `json:"num_queries"`
+}
+
+func init() {
+	resource.RegisterService(svision.API, model, resource.Registration[svision.Service, *DistanceDetectorConfig]{
+		DeprecatedRobotConstructor: func(ctx context.Context, r any, c resource.Config, logger golog.Logger) (svision.Service, error) {
+			attrs, err := resource.NativeConfig[*DistanceDetectorConfig](c)
+			if err != nil {
+				return nil, err
+			}
+			actualR, err := utils.AssertType[robot.Robot](r)
+			if err != nil {
+				return nil, err
+			}
+			return registerObstacleDistanceDetector(ctx, c.ResourceName(), attrs, actualR)
+		},
+	})
+}
+
+// Validate ensures all parts of the config are valid.
+func (config *DistanceDetectorConfig) Validate(path string) ([]string, error) {
+	var deps []string
```

Comment:
Right now, this is a nil pointer to a string array. You can return an empty slice, which requires you to make it, i.e. deps := make([]string,0) or deps := []string{}